youboraplugin/src/main/java/com/kaltura/playkit/plugins/youbora/PKYouboraAdsAdapter.java

```diff
@@ -246,7 +261,10 @@ private void addListeners() {
             lastReportedAdBitrate = currentAdInfo.getMediaBitrate();
             printLastReportedAdPlayhead();
             fireStart();
-            fireJoin();
+            Map<String, String> adIdMap = getAdIdMap(currentAdInfo);
+            if (adIdMap != null) {
+                fireManifest(adIdMap);
```

Comment:
Check if fireManifest is being fired if we don't call it

Reply:
/adManifest network call anyways will happen. Difference is that now we are sending adId as well.

Reply:
2 more scenarios:

1. If I don’t go to background during the playback session then ‘adId’ is being sent in fireManifest network call. And later no fireManifest is fired. However we are firing fireManifest everytime but it gets dropped because I can see in youbora sdk there is flag isAdsManifestSent which does not sent the network call again and again.
 

2. If I go to background any time before the ad start, I can see another fireManifest network call is fired which seems fine BUT there is no ‘adId’ param being sent with network call.

Reply:
> 2 more scenarios:
> 
>     1. If I don’t go to background during the playback session then ‘adId’ is being sent in fireManifest network call. And later no fireManifest is fired. However we are firing fireManifest everytime but it gets dropped because I can see in youbora sdk there is flag isAdsManifestSent which does not sent the network call again and again.
> 
>     2. If I go to background any time before the ad start, I can see another fireManifest network call is fired which seems fine BUT there is no ‘adId’ param being sent with network call.

Just check if the AdId is actually there in this case.